terraform/azure/traffic/main.tf

```diff
@@ -0,0 +1,74 @@
+variable "AZURE_APP_ID" {
+  description = "Azure app ID"
+}
+
+variable "AZURE_PASSWORD" {
+  description = "Azure AZURE_PASSWORD"
+}
+
+variable "DEPLOYMENT_NAME" {
+  description = "Name of deployment - used for the cluster name.  Example: rotate"
+}
+
+variable "VOTE_URL" {
+  description = "Vote app url"
+}
+
+variable "RESULT_URL" {
+  description = "Result app url"
+}
+
+locals {
+  cluster_name = "compute-traffic-${var.DEPLOYMENT_NAME}"
+}
+
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "2.66.0"
+    }
+  }
+
+  required_version = ">= 0.14"
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "traffic" {
+  name     = "${local.cluster_name}-resources"
+  location = "West US 2"
+}
+
+module "linuxservers" {
```

Comment:
Ran into a couple issues on the creation of this:

```│ Error: Invalid function argument
│
│   on .terraform/modules/linuxservers/main.tf line 109, in resource "azurerm_virtual_machine" "vm-linux":
│  109:         key_data = file(ssh_keys.value)
│     ├────────────────
│     │ ssh_keys.value is "~/.ssh/id_rsa.pub"
│
│ Invalid value for "path" parameter: no file exists at ~/.ssh/id_rsa.pub; this function works only with files
│ that are distributed as part of the configuration source code, so if this file will be created by a resource in
│ this configuration you must instead obtain this result from an attribute of that resource.
```

And 

```
│ Error: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter" Message="Required parameter 'adminPassword' is missing (null)." Target="adminPassword"
│
│   with module.linuxservers.azurerm_virtual_machine.vm-linux[0],
│   on .terraform/modules/linuxservers/main.tf line 32, in resource "azurerm_virtual_machine" "vm-linux":
│   32: resource "azurerm_virtual_machine" "vm-linux" {
```

To resolve I made this change to azure/traffic/main.tf:

```
diff --git a/terraform/azure/traffic/main.tf b/terraform/azure/traffic/main.tf
index 2e6fd94..1101437 100644
--- a/terraform/azure/traffic/main.tf
+++ b/terraform/azure/traffic/main.tf
@@ -48,6 +48,8 @@ module "linuxservers" {
   vm_os_simple        = "UbuntuServer"
   public_ip_dns       = ["${local.cluster_name}vmips"]
   vnet_subnet_id      = module.network.vnet_subnets[0]
+  enable_ssh_key = false
+  admin_password = "<password>"

   depends_on = [azurerm_resource_group.traffic]
```

Reply:
It's possible the reason this happened was that there was a new rev of the `azurerm` provider released about 6 days ago, so potentially we saw different behavior because of that. We could pin our versions to prevent that.